packages/uix-core/src/cross-realm-object.ts

```diff
@@ -87,7 +89,9 @@ async function setupApiExchange<T>(
       );
     }),
     tunnel.config.timeout,
-    () => tunnel.destroy()
+    (e) => {
+      tunnel.abort(e);
+    }
```

Comment:
Now that `.destroy()` is part of the normal lifecycle, this timeout error callback should call `tunnel.abort(e)`, which unregisters listeners.